script/script.js

```diff
@@ -184,6 +187,11 @@ var Vector2 = (function() {
         ctx.stroke();
     };
 
+    // Send packet to set player's direction on a delay, to avoid running in to rate-limit.
+    setInterval(function() {
+        sendPacket(targetDirection);
+    }, 55);
+
```

Comment:
I think the issue is that if you're ever sending packets faster than 55ms (presumably with some tolerance, that has been keeping us in for 30s or so) then you get disconnected. So sending a packet with the most up-to-date direction every 55ms is the best we can possibly do, short of taking advantage of any tolerance they give us, no? I might be misunderstanding what triggers the disconnect though...

At any rate, I'm not entirely convinced the AI should need to change the direction more than ~20 times per second - it still is "reacting" faster than this, in the sense that it can still update its target direction as frequently as it wants based on reading the state.